docs/enhancements/install-strategy.md

```diff
@@ -0,0 +1,196 @@
+  * SSHKnownHosts
+  * InstallEnv: Too specific to openshift-install.
+  * InstallConfigSecretRef: One of the more contentious parts of the current API.
+  * ReleaseImage: Legacy Hive feature, use ClusterImageSet instead.
+
+All conditions should initialize to Unknown status as soon as possible, and then be set to True/False once known.
+
+Condition | Description
+--------- | -----------
+Ready | Important overall condition indicating if the cluster is ready or not. Reason and Message provide as much detail as possible for the user. Message could contain attempt x of y if the install strategy support retries.
+RequirementsMet | True once we are ready to launch the install. False if we hit a problem and can't start. Reasons: ClusterImageSetNotFound, etc.
+Failed | True once an install attempt has failed. Remains true if retries are in play even if another attempt is in progress. Move to False once Ready goes to True. Reason/Message should contain details on what happened if it's possible to detect, otherwise Unknown. *TODO*: Unfortunate naming here, this matches current ProvisionFailed condition, which means a failure has been encountered, not necessarily that we've given up. Failing would be a better term, but it's risky to change the ProvisionFailed condition meaning on ClusterDeployment when we transfer. Thoughts?
+Retrying | Unknown initially, True if we've hit an error and are retrying, False if we've given up or don't support retries. Message should indicate attempt x of y.
+
+These interface conditions will be copied back onto ClusterDeployment by Hive controllers with a "Provision" prefix, the install strategy controller does not need to worry about this and in general should never write to ClusterDeployment. example: Failed becomes ProvisionFailed (which exists today on ClusterDeployment for this purpose)
+
+Install strategy controllers can add their own conditions, but these will not transfer back to the ClusterDeployment. UIs around this process may need to show install strategy conditions explicitly.
+
+The core install strategy CRDs and controllers discussed in this document should live in Hive. However, in theory, it would be possible for an external application to implement it's own install strategy outside of Hive and still interface with ClusterDeployment using this system.
+
+### Migration Plan
+
+No breakage for current installation API is planned. (with exception of as yet unreleased Agent install strategy)
+
+Initial goal would be to breakout ClusterDeployment.Spec.InstallStrategy.Agent. The only implementation here is agent based and this is hidden behind an alpha feature gate, and not yet released to customers. Replace this with a new hive.openshift.io.AgentInstallStrategy CRD, and a ClusterDeployment.Spec.InstallStrategyRef link. Refactor the Agent controllers (currently living in the assisted service repo, coming to hive in future) to watch this resource instead of ClusterDeployment.
```

Comment:
Just stating the obvious: these changes would need to happen real soon for inclusion in the first release to customers of the agent-based install tooling.

Reply:
I was worried about that. How soon is real soon. :) We can get cracking fast but will need a little time on Hive side. The Agent controller changes should be fairly light.